pkg/client/simple/vfsclientset/clientset.go

```diff
@@ -154,7 +172,46 @@ func DeleteAllClusterState(basePath vfs.Path) error {
 	return nil
 }
 
+func deleteAllPaths(basePath vfs.Path) error {
+	paths, err := basePath.ReadTree()
+	if err != nil {
+		return fmt.Errorf("error listing files in state store: %v", err)
+	}
+
+	for _, path := range paths {
+		err = path.Remove()
```

Comment:
Does this need to ignore "NotFound" type errors?

Reply:
In the unfortunate scenarios of trying to delete these paths while something else is writing to them?
Could be. This function is a copy of the function above though, just not checking if the root is well-known or not (since we explicitly pass the secret/key store here)

Reply:
In the scenario of trying to delete these paths when they have been deleted by a previous (failed) run of kops.

Reply:
`ReadTree` will return all _files_ that exists within that tree, and we only give the method the root of the secret/key store. So if a failed run of kops only delete some of the files, the rest would be automatically deleted on subsequent runs.